FrontEnd/src/UserAuthentication/FirestoreDB.js

```diff
@@ -1,16 +1,112 @@
-import {database} from './FirebaseConfig';
-import {doc, setDoc} from 'firebase/firestore';
+import { database } from "./FirebaseConfig";
+import {
+  query,
+  where,
+  collection,
+  doc,
+  getDocs,
+  setDoc,
+  getDoc,
+} from "firebase/firestore";
+import { storage } from "../UserAuthentication/FirebaseConfig";
+import { getDownloadURL, ref, uploadBytes } from "firebase/storage";
+import { v4 } from "uuid";
 
-async function addUser(userCredential, firstName, lastName, userName, signUpEmail){
-    const uid = userCredential.user.uid;
+async function addUser(userId, firstName, lastName, userName, signUpEmail) {
+  const userDocRef = doc(database, "users", userId);
+  return await setDoc(userDocRef, {
+    userID: userId,
+    FirstName: firstName,
+    LastName: lastName,
+    UserName: userName,
+    Email: signUpEmail,
+  });
+}
+
+async function isDatabase() {
+  const usersCollection = query(collection(database, "users"));
+  const querySnapshot = await getDocs(usersCollection);
+  return !querySnapshot.empty;
+}
+
+async function isUsernameUnique(username) {
+  const databaseExists = await isDatabase("users");
+  if (databaseExists === false) {
+    return true;
+  }
+  const usersCollection = collection(database, "users");
+  const check = query(usersCollection, where("UserName", "==", username));
+  const querySnapshot = await getDocs(check);
+  return querySnapshot.empty;
+}
+
+async function updateUserData() {
```

Comment:
This says `updateUserData` but there's no new user data passed. Could you clarify what this function does exactly?